maven-xml-impl/src/main/java/org/codehaus/plexus/util/xml/Xpp3DomBuilder.java

```diff
@@ -62,8 +64,10 @@ public static Xpp3Dom build(Reader reader, boolean trim) throws XmlPullParserExc
      */
     public static Xpp3Dom build(Reader reader, boolean trim, InputLocationBuilder locationBuilder)
             throws XmlPullParserException, IOException {
-        return new Xpp3Dom(
-                XmlNodeBuilder.build(reader, trim, locationBuilder != null ? locationBuilder::toInputLocation : null));
+        try (Reader closeMe = reader) {
+            return new Xpp3Dom(XmlNodeBuilder.build(
+                    reader, trim, locationBuilder != null ? locationBuilder::toInputLocation : null));
+        }
     }
```

Comment:
These two hunks I don't understand.

Reply:
The `XmlNodeBuilder` is new in maven 4, so we can change it the way we want.  The `Xpp3DomBuilder` comes from `plexus-utils` but now delegates to the the new `XmlNodeBuilder`.  In order to preserve compatibility, I'd rather keep the previous behaviour for this class.

Fwiw, my plan is to release 4.0.0-alpha-5, and work on the plexus-xml / plexus-utils 4.0 split, which can then later be used in maven 4.0.0-alpha-6 to get rid of the redefined plexus classes in maven. 

Does that make more sense ?

Reply:
Yes, absolutely!